### ISSUE TITLE
Unable to locate dotnet 6 SDK on macOS

### ISSUE THREAD

**Issue by Candygoblen123** [bluemagic123/tModLoader#3365]
### Version

Other (please specify elsewhere)

### OS

Mac

### Platform

Steam

### This bug affects

Mod capability as a Modder

### Description

On an arm Mac, tModloader is unable to locate the dotnet 6 SDK. I am unable to access the "Develop Mods" menu. 

This is on 1.4.4-preview.

### Log File

```
[10:40:15.869] [Main Thread/INFO] [tML]: Save Are Located At: /Users/candy/Library/Application Support/Terraria/tModLoader-preview
[10:40:15.908] [Main Thread/INFO] [tML]: Starting tModLoader client 1.4.4.9+2023.03.85.0|2022.09|preview|Preview|e1c273f214dfd11bcf8bc62ec90ff0647fcaa464|5249844689595639374 built 3/31/2023 9:45 AM
[10:40:15.915] [Main Thread/INFO] [tML]: Log date: 3/31/2023
[10:40:15.917] [Main Thread/INFO] [tML]: Running on OSX (v13.2.1) X64 NetCore 6.0.0
[10:40:15.920] [Main Thread/INFO] [tML]: FrameworkDescription: .NET 6.0.0-rtm.21522.10
[10:40:15.920] [Main Thread/INFO] [tML]: Executable: /Users/candy/Applications/SteamLibrary/steamapps/common/tModLoader/tModLoader.dll
[10:40:15.920] [Main Thread/INFO] [tML]: Working Directory: /Users/candy/Applications/SteamLibrary/steamapps/common/tModLoader
[10:40:15.936] [Main Thread/INFO] [tML]: Developer mode enabled
[10:40:17.193] [Main Thread/DEBUG] [FNA]: Querying linked library versions...
[10:40:17.216] [Main Thread/DEBUG] [FNA]: SDL v2.24.0
[10:40:17.219] [Main Thread/DEBUG] [tML]: Native Resolve: FNA, Version=22.7.0.0, Culture=neutral, PublicKeyToken=null -> FNA3D
[10:40:17.220] [Main Thread/DEBUG] [tML]: 	attempting load /Users/candy/Applications/SteamLibrary/steamapps/common/tModLoader/Libraries/Native/OSX/libFNA3D.0.dylib
[10:40:17.222] [Main Thread/DEBUG] [tML]: 	success
[10:40:17.223] [Main Thread/DEBUG] [FNA]: FNA3D v22.8.0
[10:40:17.225] [Main Thread/DEBUG] [tML]: Native Resolve: FNA, Version=22.7.0.0, Culture=neutral, PublicKeyToken=null -> FAudio
[10:40:17.225] [Main Thread/DEBUG] [tML]: 	attempting load /Users/candy/Applications/SteamLibrary/steamapps/common/tModLoader/Libraries/Native/OSX/libFAudio.0.dylib
[10:40:17.227] [Main Thread/DEBUG] [tML]: 	success
[10:40:17.227] [Main Thread/DEBUG] [FNA]: FAudio v22.8.0
[10:40:18.226] [Main Thread/INFO] [tML]: Distribution Platform: Steam. Detection method: CWD is /steamapps/
[10:40:18.255] [Main Thread/DEBUG] [tML]: Process.Start (UseShellExecute = False): "/Users/candy/Applications/SteamLibrary/steamapps/common/tModLoader/dotnet/6.0.0/dotnet" tModLoader.dll -terrariasteamclient 225
[10:40:19.122] [Main Thread/DEBUG] [TerrariaSteamClient]: Recv: init_success
[10:40:19.127] [Main Thread/DEBUG] [TerrariaSteamClient]: Send: acknowledged
[10:40:19.196] [Main Thread/DEBUG] [tML]: ILHook Ionic.Zip.ZipEntry::ValidateOutput(string, Stream, out string) added by tModLoader
[10:40:20.023] [Main Thread/DEBUG] [tML]: Native Resolve: Steamworks.NET, Version=20.1.0.0, Culture=neutral, PublicKeyToken=null -> steam_api
[10:40:20.026] [Main Thread/DEBUG] [tML]: 	attempting load /Users/candy/Applications/SteamLibrary/steamapps/common/tModLoader/Libraries/Native/OSX/libsteam_api64.dylib
[10:40:20.028] [Main Thread/DEBUG] [tML]: 	success
[10:40:20.200] [Main Thread/INFO] [Terraria]: Steam Cloud Quota: 901.4 MB available
[10:40:20.267] [Main Thread/WARN] [FNA]: Non-standard D32S8 window depth format!
[10:40:20.271] [Main Thread/INFO] [FNA]: FNA3D Driver: OpenGL
[10:40:20.272] [Main Thread/INFO] [FNA]: OpenGL Renderer: Apple M1
[10:40:20.272] [Main Thread/INFO] [FNA]: OpenGL Driver: 2.1 Metal - 83
[10:40:20.272] [Main Thread/INFO] [FNA]: OpenGL Vendor: Apple
[10:40:20.274] [Main Thread/INFO] [FNA]: MojoShader Profile: glsl120
[10:40:20.299] [Main Thread/INFO] [tML]: Terraria Steam Install Location assumed to be: /Users/candy/Applications/SteamLibrary/steamapps/common/Terraria/Terraria.app/Contents/Resources
[10:40:21.247] [Main Thread/INFO] [tML]: Maximum Resolution is 2560 x 1600
[10:40:21.272] [Main Thread/DEBUG] [Terraria]: Device Created, Adapter: 0, DisplayMode: {{Width:1680 Height:1050 Format:Color}}, Profile: HiDef, Width: 2560, Height: 1440, Fullscreen: False, Display: \\.\DISPLAY1
[10:40:23.237] [Main Thread/DEBUG] [tML]: Native Resolve: tModLoader, Version=1.4.4.9, Culture=neutral, PublicKeyToken=null -> RzChromaSDK64.dll
[10:40:23.238] [Main Thread/DEBUG] [tML]: 	not found
[10:40:23.241] [Main Thread/DEBUG] [tML]: Native Resolve: tModLoader, Version=1.4.4.9, Culture=neutral, PublicKeyToken=null -> CUESDK.x64_2019.dll
[10:40:23.241] [Main Thread/DEBUG] [tML]: 	not found
[10:40:28.596] [9/DEBUG] [tML]: SSDP search line seperator: CRLF
[10:40:43.529] [.NET ThreadPool Worker/WARN] [tML]: Did not find enabled.json file
[10:40:43.533] [.NET ThreadPool Worker/INFO] [tML]: Finding Mods...
[10:40:43.536] [.NET ThreadPool Worker/INFO] [tML]: Constructing Mods...
[10:40:43.615] [.NET ThreadPool Worker/INFO] [tML]: Adding Content: ModLoader (tModLoader) v2023.3.85.0
[10:40:43.769] [.NET ThreadPool Worker/INFO] [tML]: Resizing...
[10:40:44.017] [.NET ThreadPool Worker/INFO] [tML]: Configuring Content: ModLoader (tModLoader) v2023.3.85.0
[10:40:44.159] [.NET ThreadPool Worker/INFO] [tML]: Finalizing Content: ModLoader (tModLoader) v2023.3.85.0
[10:40:44.214] [.NET ThreadPool Worker/INFO] [tML]: RAM usage: 442.7 MB
[10:40:44.314] [.NET ThreadPool Worker/INFO] [tML]: Adding Recipes...
[10:44:44.791] [Main Thread/DEBUG] [tML]: Process.Start (UseShellExecute = False): "which" dotnet
[10:44:44.885] [Main Thread/DEBUG] [tML]: Can't find SystemDotnetPath
[10:44:44.886] [Main Thread/DEBUG] [tML]: Process.Start (UseShellExecute = False): "dotnet" --list-sdks
[10:44:45.671] [Main Thread/INFO] [tML]: 

[10:44:45.763] [Main Thread/WARN] [tML]: Silently Caught Exception: 
System.ArgumentException: Version string portion was too short or too long. (Parameter 'input')
   at Hook<System.Void Terraria.ModLoader.Engine.LoggingHooks::Hook_StackTrace_CaptureStackTrace(Terraria.ModLoader.Engine.LoggingHooks+orig_StackTrace_CaptureStackTrace,System.Diagnostics.StackTrace,System.Int32,System.Boolean,System.Exception)>(StackTrace , Int32 , Boolean , Exception )
   at SyncProxy<System.Void System.Diagnostics.StackTrace:CaptureStackTrace(System.Int32, System.Boolean, System.Exception)>(StackTrace , Int32 , Boolean , Exception )
   at System.Diagnostics.StackTrace..ctor(Boolean fNeedFileInfo)
   at Terraria.ModLoader.Logging.FirstChanceExceptionHandler(Object sender, FirstChanceExceptionEventArgs args)
   at System.Version.ParseVersion(ReadOnlySpan`1 input, Boolean throwOnFailure)
   at System.Version.Parse(String input)
   at System.Version..ctor(String version)
   at Terraria.ModLoader.UI.UIModSources.IsCompatibleDotnetSdkAvailable()
   at Terraria.ModLoader.UI.UIModSources.ShowInfoMessages()
   at Terraria.ModLoader.UI.UIModSources.OnActivate()
   at Terraria.UI.UIElement.Activate()
   at Terraria.UI.UserInterface.SetState(UIState state)
   at Terraria.GameContent.UI.States.UIWorkshopHub.Click_OpenModSourcesMenu(UIMouseEvent evt, UIElement listeningElement)
   at Terraria.UI.UIElement.LeftClick(UIMouseEvent evt)
   at Terraria.UI.UserInterface.<>c.<.ctor>b__21_2(UIElement element, UIMouseEvent evt)
   at Terraria.UI.UserInterface.HandleClick(InputPointerCache cache, GameTime time, Boolean isDown, UIElement mouseElement)
   at Terraria.UI.UserInterface.Update_Inner(GameTime time, UIElement uIElement, Boolean& num)
   at Terraria.UI.UserInterface.Update(GameTime time)
   at Terraria.Main.UpdateUIStates(GameTime gameTime)
   at Terraria.Main.DoUpdate(GameTime& gameTime)
   at Terraria.Main.Update(GameTime gameTime)
   at Microsoft.Xna.Framework.Game.Tick()
   at Microsoft.Xna.Framework.Game.RunLoop()
   at Microsoft.Xna.Framework.Game.Run()
   at Terraria.Program.RunGame()
   at Terraria.Program.LaunchGame_(Boolean isServer)
   at Terraria.Program.LaunchGame(String[] args, Boolean monoArgs)
   at Terraria.MonoLaunch.Main_End(String[] args)
   at Terraria.MonoLaunch.<>c__DisplayClass1_0.<Main>b__0()
   at Terraria.MonoLaunch.Main(String[] args)

[10:44:45.767] [Main Thread/DEBUG] [tML]: 'dotnet --list-sdks' check failed: 
System.ArgumentException: Version string portion was too short or too long. (Parameter 'input')
   at System.Version.ParseVersion(ReadOnlySpan`1 input, Boolean throwOnFailure)
   at System.Version.Parse(String input)
   at System.Version..ctor(String version)
   at Terraria.ModLoader.UI.UIModSources.IsCompatibleDotnetSdkAvailable() in tModLoader\Terraria\ModLoader\UI\UIModSources.cs:line 329
[10:44:49.775] [Main Thread/DEBUG] [TerrariaSteamClient]: Send: shutdown
```

### Steps to reproduce

1. Install the dotnet sdk
2. launch the game
3. go to the Develop Mods menu
4. See the game prompt you to install an SDK that you already have installed.

### Expected Behavior

tModloader is able to find my dotnet sdk and I am able to compile mods with it.

### Actual Behavior

The game prompts you to install the dotnet 6 sdk that you already have installed.

### Reproduction frequency

Every time.

### Additional Information

```
~ % which dotnet
/usr/local/share/dotnet/dotnet

~ % dotnet --list-sdks
6.0.407 [/usr/local/share/dotnet/sdk]

~ % /usr/local/share/dotnet/x64/dotnet --list-sdks
6.0.407 [/usr/local/share/dotnet/x64/sdk]
```

The base `/usr/local/share/dotnet` has arm64 dotnet installed, with the x86_64 dotnet installed at `/usr/local/share/dotnet/x64/`

**Comment by Chicken-Bones** [comment]
Thankyou for all the information provided. This definitely _should_ be working for you. Relevant log lines are:
```
[10:44:44.791] [Main Thread/DEBUG] [tML]: Process.Start (UseShellExecute = False): "which" dotnet
[10:44:44.885] [Main Thread/DEBUG] [tML]: Can't find SystemDotnetPath
[10:44:44.886] [Main Thread/DEBUG] [tML]: Process.Start (UseShellExecute = False): "dotnet" --list-sdks
[10:44:45.671] [Main Thread/INFO] [tML]: 

```

As per #3334 the use of `which` is supposed to identify the actual install of `dotnet` on your `PATH` rather than the local one. It appears to be failing in tML for some reason. As you've shown, the expected output of `/usr/local/share/dotnet/dotnet` should work, but maybe the `which` command isn't working from inside tML.

A workaround you can apply in the interim, is to launch tML via `/usr/local/share/dotnet/x64/sdk/dotnet tModLoader.dll` from the command line yourself

**Comment by Candygoblen123** [comment]
I can confirm that launching tML with `/usr/local/share/dotnet/x64/dotnet tModLoader.dll` does work.

**Comment by Chicken-Bones** [comment]
I was able to replicate this on my mac. The issue is that `dotnet` is not on your `PATH` by the time steam is launched. We might be able to resolve this by launching the `which` process with `UseShellExecute = true` but I'll have to check for side-effects.

You should also be able to run `./start-tModLoader.sh` to launch for now

**Comment by Chicken-Bones** [comment]
Fixed in 468d6c8
